tests/integration/test_wazuh_db/test_wazuh_db.py

```diff
@@ -0,0 +1,569 @@
+    - manager
+
+daemons:
+    - wazuh-db
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - Debian Buster
+    - Red Hat 8
+    - Ubuntu Focal
+    - Ubuntu Bionic
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/reference/daemons/wazuh-db.html
+
+tags:
+    - wazuh_db
+'''
+from pathlib import Path
+import re
+import time
+import pytest
+import json
+import random
+
+from wazuh_testing.constants.paths.sockets import WAZUH_DB_SOCKET_PATH
+from wazuh_testing.constants.paths.logs import WAZUH_LOG_PATH
+from wazuh_testing.constants.daemons import WAZUH_DB_DAEMON
+from wazuh_testing.tools.monitors import file_monitor
+from wazuh_testing.utils.database import delete_dbs
+from wazuh_testing.utils.manage_agents import remove_all_agents
+from wazuh_testing.utils.services import control_service
+from wazuh_testing.utils.callbacks import make_callback
+from wazuh_testing.modules.wazuh_db import WAZUH_DB_PREFIX
+from wazuh_testing.utils import configuration
+
+from . import TEST_CASES_FOLDER_PATH
+
+# Marks
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+# Configurations
+t1_cases_path = Path(TEST_CASES_FOLDER_PATH, 'cases_agent_messages.yaml')
+t1_config_parameters, t1_config_metadata, t1_case_ids = configuration.get_test_cases_data(t1_cases_path)
+
+t2_cases_path = Path(TEST_CASES_FOLDER_PATH, 'cases_global_messages.yaml')
+t2_config_parameters, t2_config_metadata, t2_case_ids = configuration.get_test_cases_data(t2_cases_path)
+
+# Variables
+receiver_sockets_params = [(WAZUH_DB_SOCKET_PATH, 'AF_UNIX', 'TCP')]
+WAZUH_DB_CHECKSUM_CALCULUS_TIMEOUT = 20
+
+# mitm_analysisd = ManInTheMiddle(address=analysis_path, family='AF_UNIX', connection_protocol='UDP')
+# monitored_sockets_params is a List of daemons to start with optional ManInTheMiddle to monitor
+# List items -> (wazuh_daemon: str,(
+#                mitm: ManInTheMiddle
+#                daemon_first: bool))
+# Example1 -> ('wazuh-clusterd', None)              Only start wazuh-clusterd with no MITM
+# Example2 -> ('wazuh-clusterd', (my_mitm, True))   Start MITM and then wazuh-clusterd
+monitored_sockets_params = [(WAZUH_DB_DAEMON, None, True)]
+
+receiver_sockets, monitored_sockets, log_monitors = None, None, None  # Set in the fixtures
+
+# Test daemons to restart.
+daemons_handler_configuration = {'all_daemons': True}
+
+
+def regex_match(regex, string):
+    regex = regex.replace('*', '.*')
+    regex = regex.replace('[', '')
+    regex = regex.replace(']', '')
+    regex = regex.replace('(', '')
+    regex = regex.replace(')', '')
+    string = string.replace('[', '')
+    string = string.replace(']', '')
+    string = string.replace('(', '')
+    string = string.replace(')', '')
+    return re.match(regex, string)
+
+
+@pytest.fixture(scope='module')
+def clean_registered_agents():
+    remove_all_agents('wazuhdb')
+    time.sleep(5)
```

Comment:
Why this sleep, couldn't it be in the test?

Reply:
I suppose it is because it is a fixture, in the test I have no control to give it some time